manifests/firewalld_shim.pp

```diff
@@ -0,0 +1,137 @@
+# These items mimic components in the actual `firewalld` module but set them to
+# safer defaults per the usual "authoritative control" idea of SIMP.
+#
+# Since the `firewalld` module is designed to be Hiera-driven, this was more
+# understandable and safer than encapsulating the entire module in the
+# `iptables` module directly.
+#
+# @param enable
+#   Activate the firewalld shim capabilties.
+#
+# @param complete_reload
+#   The current firewalld module has the capability to perform a complete reload
+#   of firewalld which breaks any existing connections. This is extremely
+#   dangerous and this class overrides and disables this capability by default.
+#
+#   * Set to ``true`` to re-enable this capability.
+#
+# @param lockdown
+#   Set ``firewalld`` in ``lockdown`` mode which disallows manipulation by
+#   applications.
+#
+#   * This makes sense to do by default since puppet is meant to be
+#     authoritative on the system.
+#
+# @param default_zone
```

Comment:
Need to note that if this not 'simp', a 'simp' zone will still be created and all rules from this module will be applied to the 'simp' zone.

Reply:
Good idea.